### PR TITLE
Avatars rendered correctly based on Scene.shouldRenderAvatars

### PR DESCRIPTION
**Pull request by venkatn93**
Providing a fix for avatar rendering not turning on/off in real time when the developer menu item is checked/unchecked.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: dacc6f66c031c302264f939a4eaa0c128d3917f4 available at: https://gist.github.com/4f1088d890fb2c08e91e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5308.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5308.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bbd7b2a65e22a4f179de700bd8655cd72e9ed714 available at: https://gist.github.com/4f1088d890fb2c08e91e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5308.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5308.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f95bd3f7bd635469636533d165576c7c5834cb1e available at: https://gist.github.com/4f1088d890fb2c08e91e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5308.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5308.dmg

File: interface/src/avatar/AvatarManager.cpp [hunk 1]
```diff
@@ -310,3 +316,26 @@ void AvatarManager::updateAvatarPhysicsShape(const QUuid& id) {
         }
     }
 }
+
+void AvatarManager::updateAvatarRenderStatus(bool shouldRenderAvatars) {
```
Comment (jherico) [inline interface/src/avatar/AvatarManager.cpp]: it looks like this only gets called manually based on the connection.  Is it guaranteed to be called at the start of the application?  

Also, do you need to protect against it being called with the same state that we're currently in?

Comment (venkatn93) [inline interface/src/avatar/AvatarManager.cpp]: At the start of the program both avatar and scene rendering is defaulted to true. And this function will only be called through a signal if that bool changes. The if/else takes care of setting the rendering based on the current value.

Comment (jherico) [inline interface/src/avatar/AvatarManager.cpp]: so how does _avatarsLastInScene get populated initially?

**Comment by jherico** [comment]
Looks like there are issues when list of avatars to render should change while rendering

File: interface/src/avatar/AvatarManager.cpp (hunk 1 above)
Comment (venkatn93) [inline interface/src/avatar/AvatarManager.cpp]: Im putting up a PR that fixes avatar rendering. I have got it working for when the avatar leaves the scene or gets updated in any way when the rendering is turned off

**Comment by hifi-gustavo** [comment]
Build console output for commit: 10e0a93ac096e460078cd56790f65afe4de691d4 available at: https://gist.github.com/4f1088d890fb2c08e91e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5308.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5308.dmg

File: interface/src/avatar/Avatar.h [hunk 2]
```diff
@@ -91,6 +91,7 @@ class Avatar : public AvatarData {
     void removeFromScene(AvatarSharedPointer self, std::shared_ptr<render::Scene> scene, 
                                 render::PendingChanges& pendingChanges);
 
+
```
Comment (Atlante45) [inline interface/src/avatar/Avatar.h]: Extra line

File: interface/src/avatar/AvatarManager.cpp [hunk 3]
```diff
@@ -72,9 +72,13 @@ void AvatarManager::init() {
     _myAvatar->init();
     _avatarHash.insert(MY_AVATAR_KEY, _myAvatar);
 
+    connect(&(*DependencyManager::get<SceneScriptingInterface>()), &SceneScriptingInterface::shouldRenderAvatarsChanged, this, &AvatarManager::updateAvatarRenderStatus, Qt::QueuedConnection);
```
Comment (Atlante45) [inline interface/src/avatar/AvatarManager.cpp]: Don't do `&(*ptr)`, you can do `ptr.data()` and it'll give you the raw pointer directly.

File: interface/src/avatar/AvatarManager.cpp [hunk 4]
```diff
@@ -310,3 +316,24 @@ void AvatarManager::updateAvatarPhysicsShape(const QUuid& id) {
...
+    if (DependencyManager::get<SceneScriptingInterface>()->shouldRenderAvatars()) {
+        for (auto avatarData : _avatarHash) {
+            auto avatar = std::dynamic_pointer_cast<Avatar>(avatarData);
+            render::ScenePointer scene = Application::getInstance()->getMain3DScene();
+            render::PendingChanges pendingChanges;
+            avatar->addToScene(avatar, scene, pendingChanges);
+            scene->enqueuePendingChanges(pendingChanges);
+        }
+    }
+    else {
```
Comment (Atlante45) [inline interface/src/avatar/AvatarManager.cpp]: coding standard: '} else {'

File: interface/src/avatar/AvatarManager.h [hunk 5]
```diff
@@ -88,6 +89,7 @@ public slots:
     SetOfMotionStates _motionStatesToAdd;
     VectorOfMotionStates _motionStatesToDelete;
     VectorOfMotionStates _tempMotionStates;
+
```
Comment (Atlante45) [inline interface/src/avatar/AvatarManager.h]: extra line

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1f1a5f8e751b3acb810be81e833984762bfbd23c available at: https://gist.github.com/4f1088d890fb2c08e91e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5308.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5308.dmg

**Comment by venkatn93** [comment]
Fixed all the above issues. PR is now ready to merge

**Comment by hifi-gustavo** [comment]
Build console output for commit: ef3c6b25e0d8547d2dae1f81ede4ef505c246a3b available at: https://gist.github.com/4f1088d890fb2c08e91e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5308.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5308.dmg

**Pull request merged**